src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![deny(rust_2018_idioms)]
```

Comment:
Please make this a `warn`, I don't think I want this to fail when I am working locally. if you want you can make CI use `-D warnings`.

Also please move this down below the feature gate, to where there already is an `allow`.

Reply:
Oh, and if you intend to deny warnings on CI, please make that a separate PR.